prow/gerrit/client/client_test.go

```diff
@@ -48,7 +52,99 @@ func (f *fgc) ListChangeComments(id string) (*map[string][]gerrit.CommentInfo, *
 	}
 
 	return &comments, nil, nil
+}
+
+func TestApplyGlobalConfigOnce(t *testing.T) {
+	// Empty opener so *syntTime won't panic.
+	dir := t.TempDir()
+	path := filepath.Join(dir, "value.txt")
+	opener, err := io.NewOpener(context.Background(), "", "")
+	if err != nil {
+		t.Fatalf("Failed to create opener: %v", err)
+	}
+
+	// Fixed org/repo, as this test doesn't really check the output.
+	cfg := config.Config{
+		ProwConfig: config.ProwConfig{
+			Gerrit: config.Gerrit{
+				OrgReposConfig: &config.GerritOrgRepoConfigs{
+					{
+						Org:   "foo1",
+						Repos: []string{"bar1"},
+					},
+				},
+			},
+		},
+	}
+
+	// A thread safe map for checking addtional func.
+	var mux sync.RWMutex
+	records := make(map[string]string)
+	setRecond := func(key, val string) {
+		mux.Lock()
+		defer mux.Unlock()
+		records[key] = val
+	}
+	getRecord := func(key string) string {
+		mux.RLock()
+		defer mux.RUnlock()
+		return records[key]
+	}
+
+	tests := []struct {
+		name                string
+		orgRepoConfigGetter func() *config.GerritOrgRepoConfigs
+		lastSyncTracker     *SyncTime
+		addtionalFunc       func()
+		expect              func(t *testing.T)
+	}{
+		{
+			name: "base",
+			orgRepoConfigGetter: func() *config.GerritOrgRepoConfigs {
+				return cfg.Gerrit.OrgReposConfig
+			},
+			lastSyncTracker: NewSyncTime(path, opener, context.Background()),
+			addtionalFunc: func() {
+				setRecond("base", "base")
+			},
+			expect: func(t *testing.T) {
+				if got, want := getRecord("base"), "base"; got != want {
+					t.Fatalf("Output mismatch. Want: %s, got: %s", want, got)
+				}
+			},
+		},
+		{
+			name: "nil-lastsynctracker",
+			orgRepoConfigGetter: func() *config.GerritOrgRepoConfigs {
+				return cfg.Gerrit.OrgReposConfig
+			},
+			addtionalFunc: func() {
+				setRecond("nil-lastsynctracker", "nil-lastsynctracker")
+			},
+			expect: func(t *testing.T) {
+				if got, want := getRecord("nil-lastsynctracker"), "nil-lastsynctracker"; got != want {
+					t.Fatalf("Output mismatch. Want: %s, got: %s", want, got)
+				}
+			},
+		},
+		{
+			name: "nil-addtionalfunc",
+			orgRepoConfigGetter: func() *config.GerritOrgRepoConfigs {
```

Comment:
Nit: How about explicitly setting `additionalFunc: nil` here as a field? That way I don't have to jump back up to the struct definition to see what it's supposed to default to.

But also it's not clear to me from looking at just this test case what the expectation is. Maybe a simple `expect: ...` field would help here?

Reply:
done